### ISSUE TITLE
0.625m Probes overheat with the 0.625m Heat Shield

### ISSUE THREAD

**Issue by roninpawn** [PorktoberRevolution/ReStocked#1004]
Okay. I went ALL the way down the rabbit hole with this one. After a whole day's tinkering, I've both pinpointed the problem, and solved it. So...

**The Problem**
The OCTO probe core overheats to near-explosion on descent from a standard Low Kerbin Orbit, with a 33.5km Pe. This occurs when the craft consists of simply: The 0.625m Heat Shield, the Probe Core, a 0.625m battery, and a standard parachute. On deeper dive, the other probe cores in the 0.625m class also suffer some less severe changes to their heat profiles, in similar configurations.

My testing shows that from LKO return with the stock parts, the above-described craft should not show a heat tick at all on descent. But in ReStock, the tick appears at 52km. The OCTO2, in a similar configuration (but with reaction wheels added) shows its first heat tick between 50-52km with the stock parts. But with ReStock, it begins overheating at 56km.

**Why**
The stock CUBE_MESH for these probe cores has not been included in the restock-probes.cfg. So KSP generates a mesh for each of them, based on ReStock's 3D models. The OCTO suffers most because its model includes two little nubs protruding from its bottom, which penetrate the Heat Shield. (which makes for a nice mounting-point visual) Because of these nubs, when KSP generates the cube mesh for the OCTO, it ends up (I presume) extending the mesh passed (or flush with) the heat shield's mesh. This leaves the OCTO significantly unoccluded during re-entry, and causes it to begin overheating situations where the stock part wouldn't even display a tick.

**What's the Fix?**
The fix is to simply include the stock CUBE_MESH for each of the 0.625m probe cores, as lifted from the PartsDatabase.cfg at runtime. Which I have done. And packed into a patch-mod that overwrites the restock-probes.cfg and empties the PartsDatabase.cfg to force a refresh of the cube meshes, under all circumstances.

**Oh yeah? Well... Prove it, then!**
I've created a testbed from a clean 1.12.5 Creative mode save. Extract the zip to your KSP directory and you'll find a new 'ReStock Probes Fix' save available. Load in and you've got some options. In the VAB (and there's one waiting on the launch pad) I've created a craft to test all three probe cores at once. See the ship's description for simple reproduction steps.

The other way to go is to skip the VAB and the launch pad, and just hit F9. You'll find a quicksave named 'Before Decoupling.' Load that up and you'll find the ship I mentioned waiting in orbit, on a 33.555km descent. Disable SAS and tap left to give the ship a little roll. Then hit spacebar and immediately go into timewarp. That'll decouple all three probes. And being in timewarp will let them pass right through the ablator's platforms, exiting in a perfect diagonal row. From there, warp down to 72km, put all of the probes onto their surface retrograde, _(rotating through the ships using [ and ])_ disable SAS on all the probes, and just watch.

I found it convenient to rotate through all three craft once every 1000 meters, to see when each probe core started displaying a heat tick. And F11 for heat-view made everything a lot more legible.

You'll find both my patch for the restock-probes.cfg, and the testbed I used under 'Releases' in this repository:
https://github.com/roninpawn/ReStock-0.625m-Probes-Heatshields-FIX

**List of mods installed:**
None. Then ReStock. Then my patch for ReStock.
I installed an untouched copy of KSP 1.12.5 and ran the test to get the Stock parts heat-tick heights. Then I installed JUST the ReStock directory and modulemanager from the 1.4.3 Release here on Github, and ran the tests again, evidencing the changes to the OCTO and OCTO2's heat profiles. Then I unzipped my patch, and ran the tests again. Which returned all results to the same as Stock.

**How did you even find this?**
The reason this ended up mattering enough for me to track it down was that I've begun playing a career mode using the mod 'Probes Before Crew.' That probably says it all, huh? I've had reasons to return probes in some of the simplest configurations imaginable. And when I started dropping stuff back to Kerbin from Minmus - using the OCTO - I lost several tufts of hair to the frustration of failed returns, no matter how high I started the Pe, nor how many loops I sat there waiting for it do before coming down.

It's nearly game-breaking in Probes Before Crew, I'd say. Although I have to acknowledge that, after all my testing, I'm not at all happy to find that Squad left the stock HECS and OCTO2 in as poor a state (for high returns) as this minor oversight in ReStock. I can't imagine ever having realized that I CAN return the OCTO from... probably Eeloo if I want! But the HECS is likely combust coming home from the Mun!

**Comment by Poodmund** [comment]
Hi @roninpawn, I have looked into this and compared the PartsDatabase.cfg file distributed by Steam for KSP1 1.12.5 and also the PartsDatabase.cfg file that is generated by KSP when no file is available (essentially a regen of the drag cubes) and have copied over the drag cubes from the version distributed by Steam (as there were only very minor differences and I assume this to be master) to the 0.625m Heatshield, the Probe Core parts (except the Rovermate) and also the radial, foil fuel tanks. I have pushed these changes to the 'drag-cubes' branch via the https://github.com/PorktoberRevolution/ReStocked/commit/6d3e1d0214313095a8c3d97f38fe59d6c64ff75a commit. Are you able to download these changes and test them to confirm this is fixed with these changes?

**Comment by roninpawn** [comment]
Setting up a testbed, now.

**Comment by roninpawn** [comment]
**Method:**
For this test I unzipped a fresh copy of KSP 1.12.5; Downloaded the _ReStocked-drag-cubes.zip_ from the _drag-cubes_ Github branch; And copied the _Distribution/Restock_ directory therein, into KSP's _/GameData_ directory along with a fresh download of _ModuleManager.4.2.3.dll_. I then placed the same '_Restock Probes Fix_' save file that I used as a test bed last year, into KSP's _/saves_ directory, and conducted the same test I described in my first post... F9'ing into the '_Before Decouple_' quicksave, and dropping the 3 test craft from LKO with a 33.5km Pe. I did this twice.

**Results:**
This time around the ReStock versions of the 3 probes show their first heat ticks at:

```
HECS:  52-54km
OCTO2: 50-52km
OCTO:  No tick
```
Which perfectly matches the Stock observations of these parts that I made last year.
Looks like this branch solves the probes!
